### PR TITLE
🎨 Add ability to list custom and built-in themes

### PR DESCRIPTION
**Pull request by grtcdr**
`--list-themes` can tell the user if `macchina` was able to read their custom theme and also prints built-in themes.

image

### CODE REVIEW

**Comment by uttarayan21** [comment]
I just tried and it doesn't seem to work with multiple custom themes.

Screenshot from 2021-05-17 21-39-51

**Comment by grtcdr** [comment]
> I just tried and it doesn't seem to work with multiple custom themes.
> 
> Screenshot from 2021-05-17 21-39-51

Ah damn it, does `find()` stop at the first occurrence?

**Comment by uttarayan21** [comment]
Yes you can use filter() .

**Comment by grtcdr** [comment]
> Yes you can use filter() .

Any refactoring to be done?

File: src/main.rs [hunk 1]
```diff
@@ -185,6 +187,36 @@ fn select_ascii(small: bool) -> Option<Text<'static>> {
...
+            let custom_themes = entries
+                .iter()
+                .filter(|&x| {
+                    if let Some(ext) = libmacchina::extra::path_extension(&x) {
+                        return ext == "json";
+                    }
+
+                    return false;
+                })
+                .into_iter();
```
Comment (uttarayan21) [inline src/main.rs:208]: This is a useless conversion and we don't need `into_iter()`.

File: src/main.rs [hunk 2]
```diff
@@ -185,6 +187,36 @@ fn select_ascii(small: bool) -> Option<Text<'static>> {
...
+        let entries = libmacchina::extra::list_dir_entries(&dir.join("macchina/themes"));
+        if !entries.is_empty() {
+            let custom_themes = entries
+                .iter()
+                .filter(|&x| {
+                    if let Some(ext) = libmacchina::extra::path_extension(&x) {
+                        return ext == "json";
+                    }
+
+                    return false;
```
Comment (uttarayan21) [inline src/main.rs:206]: Here 
```rust
if let Some(ext) = libmacchina::extra::path_extension(&x) {
    ext == "json"
} else {
    false
}
```
Will be more idiomatic rust.

**Review by uttarayan21 — COMMENTED** [review]
Aside from the mentioned everything else looks great :)

**Comment by uttarayan21** [comment]
Also in 
```rust
for i in 0..conflicts.len() {
    println!("• {}", conflicts[i]);
}
```
wouldn't 
```rust
for conflict in conflicts {
    println("• {}", conflict);
}
```
be better ?

File: src/main.rs (hunk 2 above)
Comment (grtcdr) [reply src/main.rs:206]: Yeah, that's cleaner.

File: src/main.rs (hunk 1 above)
Comment (grtcdr) [reply src/main.rs:208]: You're absolutely right :)

**Review by uttarayan21 — APPROVED** [review]
Looks good :)

**Pull request merged**